common/utils/retry_test.go

```diff
@@ -0,0 +1,93 @@
+package utils
+
+import (
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/quintilesims/layer0/common/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRetryTimeout(t *testing.T) {
+	start := time.Now()
+	fn := func() (bool, error) {
```

Comment:
No need for randomness here; just keep it simple and have this function sleep longer than the timeout